File: Dong/c/Queue/Queue/Queue.c
/*************************************************************************
 > File Name: Queue.c
 > Author:    DongZePing
 > Mail:      1939485575@qq.com
 > Time:      2019-07-30
 >
 > 1）顺序队列--存在假溢出现象
 > 2）支持入队和出队访问
 > 3）其它功能
 ************************************************************************/
#include "Queue.h"
#include <stdio.h>
void Queue_Init(struct Queue * queue)
{
	queue->front = 0;
	queue->rear = 0;
	queue->size = 1;
	queue->data = (int*)malloc(sizeof(int)*queue->size);
}
void inQueue(struct Queue * queue, int data)
{
	//队列满了,自动扩容
	if (queue->rear == queue->size)
	{
		int temp;
		temp = queue->size;
		queue->size = queue->size * 2;
		int *q = (int*)malloc(sizeof(int)*queue->size);
		for (int i = queue->front; i < queue->rear; i++)
		{
			q[i] = queue->data[i];
		}
		free(queue->data);
		queue->data = NULL;
		queue->data = q;
		queue->data[queue->rear++] = data;
	}
	else {
		queue->data[queue->rear++] = data;
	}
}
int outQueue(struct Queue * queue)
{
	if (queue->front == queue->rear)
	{
		return 999;
	}
	else {
		return queue->data[queue->front++];
	}
}
void Queue_Print(struct Queue * queue)
{
	for (int i = queue->front; i < queue->rear; i++)
	{
		printf("%d ", queue->data[i]);
	}
	printf("\n");
}

File: Dong/Java/src/Queue/Queue.java
package Queue;

public class Queue<T> {
	private int front;
	private int rear;
	private int size;
	private T [] data;
	//无参构造
	public Queue()
	{
		this.front = 0;
		this.rear = 0;
		this.size = 1;
		this.data = (T[])new Object[this.size];
	}
	//有参构造
	public Queue(int size)
	{
		this.front = 0;
		this.rear = 0;
		this.size = 1;
		this.data = (T[])new Object[this.size];
	}
	//入队列
	public void inQueue(T data)
	{
		if (this.rear == this.size)
		{
			int len = this.size;
			this.size = 2 * this.size;
			T [] temp = (T[])new Object[this.size];
			for (int i = 0; i < len; i++)
			{
				temp[i] = this.data[i];
			}
			this.data = temp;
			this.data[this.rear++] = data;
		}
		else {
			this.data[this.rear++] = data;
		}
	}
	//出队列
	public T outQueue()
	{
		if (this.front == 0)
		{
			return null;
		}
		else {
			return this.data[this.front++];
		}
	}
	//输出队列
	public void Queue_Print()
	{
		for (int i = this.front; i < this.rear; i++)
		{
			System.out.print(this.data[i]+" ");
		}
		System.out.println();
	}
}
